Generating the short description on the product editor

joelclimbsthings opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Automatically generating the short description after generating the long description.

Closes #39151 .

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout branch.
2. Navigate to Products -> Add new on your dev site.
3. Enter a title, and then generate a description.
4. Observe that after the long description is generated, a short description is automatically generated as a summary.

<!-- End testing instructions -->

coreymckrill committed:
WC Tracker: Add unit test for plugin feature compat data

In #38849 a change was made that allowed the class mocking necessary
for this test to happen in a way that wouldn't interfere with other
unit tests.

Fixes #38720

coreymckrill committed:
Add changelog file

anastas10s-afk committed:
Free shipping handling of minimum order rule

Changes when coupons added and prices are entered and displayed inclusive of tax

PanosSynetos committed:
Change default WooCommerce color

PanosSynetos committed:
Blankstate buttons follow theme colors

PanosSynetos committed:
WooCommerce message use admin theme color

PanosSynetos committed:
Input toggles use admin theme color

PanosSynetos committed:
Email status use admin theme color

PanosSynetos committed:
Breadcrump up link use admin theme color

PanosSynetos committed:
Importer/Exporter use admin theme color

PanosSynetos committed:
Auth page uses Woo colors, according to branding

PanosSynetos committed:
Untested extensions modal uses theme colors

PanosSynetos committed:
Use page-title-action instead of deprecated add-new-h2

PanosSynetos committed:
Shipping methods empty state follow theme colors (and minor improvements)

PanosSynetos committed:
Use new branding colors in activation and wc setup

PanosSynetos committed:
Fix activation css to use wp-admin-theme-color

PanosSynetos committed:
Remove hardcoded text color wherever variable background color is used

PanosSynetos committed:
Applied review comments

PanosSynetos committed:
Remove styles from buttons, as theme colors.css is taking care of them

PanosSynetos committed:
Changelog

PanosSynetos committed:
Changelog (again)

PanosSynetos committed:
Changelog

PanosSynetos committed:
Finally, that's the changelog :)

nielslange committed:
Link code snippet

nielslange committed:
Add snippet for chaning the number of related products output

joelclimbsthings committed:
Refactoring tiny utilities to support different editors

joelclimbsthings committed:
Prototype for generating short description after long description

joelclimbsthings committed:
Adding changelog

joelclimbsthings committed:
Print errors on short description to correct tiny editor

vedanshujain committed:
Limit index length to 191 characters by default.

Additionally, connect verify db tooling to order tables when they are enabled.

vedanshujain committed:
Coding standard fixes + changelog.

PanosSynetos committed:
Components empty content icon color

PanosSynetos committed:
Store alerts update - use admin theme colors

PanosSynetos committed:
Hover on orders view icon, use admin theme color

PanosSynetos committed:
Changelog

PanosSynetos committed:
Update helper css to use admin theme color

nima-karimi commented on plugins/woo-ai/src/product-description/product-description-button-container.tsx:
```diff
@@ -102,6 +103,18 @@ export function WriteItForMeButtonContainer() {
+		feature: WOO_AI_PLUGIN_FEATURE_NAME,
+		onStreamMessage: ( content ) => shortTinyEditor.setContent( content ),
+		onStreamError: ( error ) =>
+			shortTinyEditor.setContent( getApiError( error ) ),
```
It might be a good idea to move away from setting the content to the error message. I know it's a bit out of scope for this PR, but I think showing the error message separately in the UI would be better. Let's also do this for the main description generator if you agree.

joelclimbsthings replied:
@nima-karimi Yup, I definitely agree, although I'm uncertain of how/where to display these messages in the UI. The legacy product editor feels fairly clumsy to modify in those ways. If we end up going towards the modal [as Jarek has designed](https://github.com/woocommerce/woocommerce/issues/38810) then that will give us some better options, so perhaps we should see how we bottom out on that project? I'd be up for creating an issue around this regardless.

jarekmorawski replied:
Could we use a snackbar as a transient solution?

tommyshellberg replied:
In my [draft PR for the branding work](https://github.com/woocommerce/woocommerce/pull/39253), I'm catching errors and showing them via a snackbar. 
Specifically, using the Notices module, like so:
https://developer.wordpress.org/block-editor/reference-guides/data/data-core-notices/#createerrornotice
I think it would be nice for us to align on this approach, if it sounds good enough. @jarekmorawski

nima-karimi commented on plugins/woo-ai/src/product-description/product-description-button-container.tsx:
```diff
@@ -102,6 +103,18 @@ export function WriteItForMeButtonContainer() {
+			shortTinyEditor.setContent( getApiError( error ) ),
+		onCompletionFinished: ( reason, content ) => {
+			if ( reason === 'finished' ) {
+				shortTinyEditor.setContent( content );
```
We might want to add an indicator or a message on the UI that shows we are generating the short description too, and when it's finished it should indicate that we've modified the short description. Because it all happens automatically, the user might not be aware of the new changes.

Additionally, I think the feedback box should now appear after both descriptions are generated.

joelclimbsthings replied:
> We might want to add an indicator or a message on the UI that shows we are generating the short description too, and when it's finished it should indicate that we've modified the short description. Because it all happens automatically, the user might not be aware of the new changes.

I don't disagree, although again I'm uncertain how this would show up. I'm not sure it's worth complicating the UI or code all that much, as the short description is really just a "bonus" summary of the long description. As mentioned in my other comment I think moving to a modal would clarify this kind of thing.

> Additionally, I think the feedback box should now appear after both descriptions are generated.

I thought about this but I decided to just leave it as it for now, although it's definitely debatable. This is what I was thinking:
- I think the feedback should continue to just reflect the outcome of the primary description as the short description is just a summary, and this would continue to align with how the tracks event has been capturing this up to this point.
- Often times the short description isn't actually visible in the user's viewport, so this will just look like an awkward pause where nothing is happening if we delay the feedback. 
- I still think of the short description as something of a less-critical "bonus," and doesn't need to play much of a part in this immediate interaction. They'll likely  just notice it when continuing down the page as they create their product.

jarekmorawski replied:
I don't know how feasible it is, but it'd be nice to show a small button fixed at the bottom of the screen once the summary is generated. Clicking it would scroll down to the short description field, letting the user know there's new content they have yet to see. It could look like this:

<img width="330" alt="image" src="https://github.com/woocommerce/woocommerce/assets/79307566/d3eccf21-2360-452f-aa5e-2030afb30433">

We could then show a spotlight message to inform them how the description was generated (it'd only show up once, so the user isn't bugged with it every time they add a product). It could show up when the short description field comes into view.

<img width="818" alt="image" src="https://github.com/woocommerce/woocommerce/assets/79307566/8a741641-f433-46dd-a21e-febf05f5e148">

A low-effort version of this would show a snackbar mentioning the two descriptions. It'd only be shown after both descriptions are generated.

<img width="417" alt="image" src="https://github.com/woocommerce/woocommerce/assets/79307566/b1de8b83-f609-43d3-9ffa-4a1f17e8f15a">

As for modifying existing descriptions, I'd be wary of that. Can we not generate the short description if there's already one? In that case, it makes sense to consider it while generating the long description. I'm mentioning this because many merchants use the short description _instead_ of the long description because that's how their themes are structured.

> I still think of the short description as something of a less-critical "bonus," and doesn't need to play much of a part in this immediate interaction. They'll likely just notice it when continuing down the page as they create their product.

Can you show me what this looks like currently?

joelclimbsthings replied:
Thanks @jarekmorawski , I really like it! As it stands we've just been overwriting any description when the user requests another one via the "Write with AI" button, as well as printing errors to the editor if they occur. This is what it looks like:
![image](https://github.com/woocommerce/woocommerce/assets/444632/5067e01d-40f4-44af-af1d-d592abf542ff)


Although I've actually just made a change to display errors via the snackbar instead, at least temporarily until we can implement something like the above.

tommyshellberg replied:
I've added an issue for adding a "write it for me" button for the short description area as well as generating a notice:
https://github.com/woocommerce/woocommerce/issues/39580

nima-karimi reviewed (commented):
Works great 💯 Just added two (mostly UX-related) comments.

AndrewJDawes committed:
Fixes #38757 - pass time values on CSV importer

Validate datestrings and convert Unix timestamps to datestrings with UTC timezone specified.

joelclimbsthings committed:
Fixing error handling after rebase

joelclimbsthings committed:
Tweak to prompt to make the short description shorter

joelclimbsthings committed:
Adding package changelog

joelclimbsthings commented:
Thanks for the review @nima-karimi ! I largely agree with your feedback (see comments above), although we may want to create separate issues around those.

martynmjones committed:
Add activation option workaround to trigger `woocommerce_newly_installed` action (#38694)

* Use option to track new WC installs

* Correct hook used for disabling legacy coupon menu

* Add changelog

* Fix deprecated hook version number

* Update unit tests

* Use AccessiblePrivateMethods to register action

* Update comments

---------

Co-authored-by: Martyn Jones <martyn.jones@automattic.com>

psealock committed:
Package Release: Fix fetch depth so Changelogger has history (#39239)

set fetch depth to 0

alopezari committed:
Added new GitHub Actions workflow to arrange PR testing scrub sessions (#39142)

* Added new GitHub Actions workflow to arrange PR testing scrub sessions

* Minor fixes from PR review.

---------

Co-authored-by: Rodel Calasagsag <rodel.calasagsag@automattic.com>

jonathansadowski committed:
Update post-merge workflow to add 'needs: analysis' label (#39277)

samueljseay committed:
Add linting to CI for monorepo utils, add to CI unit tests for docs. (#39240)

nima-karimi commented:
Thanks for your comments. I still think we can address these to some degree before getting this PR out the door. My concern is that changing the short description automatically without any UI indication could confuse some users. Especially since the error message is also written directly in the content box, and the short description box is not very visible on the page. I think there's a good chance that some product descriptions might end up like "We're currently experiencing high demand for our experimental feature." without the user even noticing.

cc: @tommyshellberg @jarekmorawski if y'all have any input.

samueljseay committed:
Use syncpack customTypes to keep pnpm and node in sync. Update pnpm to 8.6.7 (#39245)

PanosSynetos committed:
Fix Storefront recommendation link and missing image in Marketplace

PanosSynetos committed:
Update count color, use admin theme

tommyshellberg commented:
> My concern is that changing the short description automatically without any UI indication could confuse some users. Especially since the error message is also written directly in the content box, and the short description box is not very visible on the page.

+1 to going away from writing error messages in the product description. I mentioned in another comment that I'm surfacing error messages through a Snackbar.
I can see it being frustrating if you already have a product description and you try to write a new one with AI only to have it fail and then overwrite your existing product description with an error message. 😞 

I would also recommend being careful about overwriting the short description. As Jarek mentioned, not only is it the displayed description for some templates, but, AFAIK, it's also often used for the subheading on search engine results which is critical for shops:
![Screen Shot on 2023-07-18 at 12:47:03](https://github.com/woocommerce/woocommerce/assets/6723003/97e24cc0-e666-4a67-adec-8f79a9f2f591)

jarekmorawski commented:
> I can see it being frustrating if you already have a product description and you try to write a new one with AI only to have it fail and then overwrite your existing product description with an error message. 😞

💯 This makes me wonder... shall we have an undo button? Or allow the user to bring back the original product description when they press `Ctrl/Cmd+Z` when either description field is in focus? 🤔

nielslange committed:
Update docs/snippets/README.md

Co-authored-by: Darren Ethier <darren@roughsmootheng.in>

nielslange committed:
Add obsolete spaces

nielslange committed:
Update docs/snippets/number-of-products-per-row.md

Co-authored-by: Darren Ethier <darren@roughsmootheng.in>

nielslange committed:
Correct safe guard function

github-actions[bot] committed:
Update changelog.txt from release 7.9.0 (#39283)

Prep trunk post release 7.9.0

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Release: Remove 8.0 change files (#39288)

Delete changelog files from 8.0 release

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Prep trunk for 8.1 cycle (#39286)

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

joelclimbsthings committed:
Minor refactor

joelclimbsthings committed:
Displaying errors via snackbar instead of in editor

joelclimbsthings committed:
Only updating short description if one doesn't already exist

joelclimbsthings committed:
Restoring UI after error is thrown on description completion

joelclimbsthings commented:
Thanks @nima-karimi @tommyshellberg @jarekmorawski 

I like the idea of using the snackbar for error messages, at least as a stop-gap until we get something more robust in place. I've implemented this in https://github.com/woocommerce/woocommerce/pull/39237/commits/4274a455f5c7bb8d4d085cfe57aca8d602a46a9b.

> This makes me wonder... shall we have an undo button? Or allow the user to bring back the original product description when they press Ctrl/Cmd+Z when either description field is in focus? 

@jarekmorawski This actually seems to be working for me in Chrome just via the native browser implementation. I seem to have to unfocus and then refocus the field, and then Ctrl-Z restores the previous description.

> I would also recommend being careful about overwriting the short description. As Jarek mentioned, not only is it the displayed description for some templates, but, AFAIK, it's also often used for the subheading on search engine results which is critical for shops:

Great point, I hadn't realized this. I'm not sure of the best solution here--I've tweaked this in https://github.com/woocommerce/woocommerce/pull/39237/commits/d2ca41caedc643fb6a5c7dcbb2484394419a7ef4 to only write the short description if one doesn't already exist. This will protect previous content, but it may result in some odd behavior when creating new products, as it will write the short description initially, and then leave the first one in place for subsequent completions. 🤔

coreymckrill committed:
Add changelog file

coreymckrill committed:
Add changelog file

coreymckrill committed:
Delete changelog file at wrong path

mattsherman committed:
Restore the woocommerce_variations_added jQuery trigger (#39301)

* Restore woocommerce_variations_added trigger

* Add e2e expect to verify that woocommerce_variations_added_functions jQuery trigger was called

vedanshujain committed:
Refactored filterd in the DB util for DRY.

PanosSynetos committed:
Wake up github actions

PanosSynetos committed:
Revert

Commit:
Fix flakiness

nerrad committed:
Add doc for unhooking/removing WooCommerce email notifications snippet. (#39280)

* add doc for unhooking/removing WooCommerce email notifications.

* Update snippet to follow snippet standards.

Co-authored-by: Niels Lange <info@nielslange.de>

* Update docs/snippets/unhook--remove-woocommerce-emails.md

Co-authored-by: Niels Lange <info@nielslange.de>

---------

Co-authored-by: Niels Lange <info@nielslange.de>

mdperez86 committed:
Create the Organization tab (#39232)

* Add Organization to the right of General tab

* Add Product catalog and Attributes sections to the Organization tab

* Add Categories and Disable product reviews blocks to Product catalog section

* Add attributes block to the attributes section

* Add changelog files

* Fix php linter error

* Remove Categories and Attributes blocks from the General tab

* Change Disable product reviews by Enable product reviews

github-actions[bot] committed:
Delete changelog files based on PR 39301 (#39315)

Delete changelog files for 39301

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

mdperez86 committed:
Unable to "turn off the new product form" from within the editor (#39317)

* Fix turn off the new block experience when tracking is turned off

* Add changelog file

Commit:
include post_ID in HPOS order edit screen

alopezari committed:
Add a new GitHub Actions workflow to track on Slack the new PRs merged into the WooCommerce repository (#39316)

Add a new GitHub Actions workflow to track on Slack the new PRs merged into the WooCommerce repository.

barryhughes committed:
Introduce grace period before asking guests to verify their email address (#39191)

* Add a grace period during which email verification is not required (order pay/conf page).

When the order confirmation (or payment) page is requested, we often want to ensure the visitor is associated with the order. However, this relies heavily on information stored in the user session and, depending on the payment gateway in use, this may not be dependable. Therefore, we've introduced a grace period during which no such verification will take place.

* Provide a mechanism for establishing server-side filters from our E2E tests.

* Make our utilities for setting up filters from E2E available in the test env.

* Update guest shopper workflow to account for order conf/payment access grace period.

* Tidy verbiage.

* Add changefile(s) from automation for the following project(s): woocommerce

* Only activate the Filter Setter (e2e utility) during e2e tests.

* Coding standard fixes for E2E utility plugin.

* e2e: Update locators for headings to use getByRole, add await to clearFilters

* e2e: Abstract the cookie domain to work on non-localhost test sites

---------

Co-authored-by: github-actions <github-actions@github.com>
Co-authored-by: Corey McKrill <916023+coreymckrill@users.noreply.github.com>

psealock committed:
Changefile from PR description: Fix community PR permissions issue (#39290)

* Add pull_request_target

* Add description to Enhancement

Commit:
Add changelog

samueljseay committed:
WooCommerce Docs: Store post meta of edit_url, refactor ManifestProcessor for testability (#39241)

Commit:
Shard unit test runs (#39302)

closes #38973

opr committed:
Update WooCommerce Blocks to 10.6.1 (#39299)

github-actions[bot] committed:
Delete changelog files based on PR 39299 (#39336)

Delete changelog files for 39299

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Delete changelog files based on PR 39317 (#39340)

Delete changelog files for 39317

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Delete changelog files based on PR 39191 (#39342)

Delete changelog files for 39191

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

nathanss committed:
[Block Product Editor]: Only run name validation if the field has been edited (#39320)

Only run name validation if the field has edit

Commit:
Revert "Shard unit test runs" (#39344)

Revert "Shard unit test runs (#39302)"

This reverts commit 9ee44947e0224c7d520946b5d79c558447a5c00a.

csmcneill committed:
Removes the "for your state" string from the checkout page if there are no current payment methods available.

jonathansadowski committed:
Abstract PR scrub action so it can be used in two places (#39329)

* Abstract PR scrub action so it can be used in two places

* Rename composite action; add newline

invalid-email-address committed:
Add changefile(s) from automation for the following project(s): woocommerce

psealock committed:
Changefile from PR description: Fix community PRs from branch named trunk (#39306)

* check if remote branch is trunk before checking out

* fix comment

Commit:
Perform a site reset if running on CI

Commit:
Run global setup

Commit:
Update API URL in workflow and run all API tests

Commit:
Update config to use API URL

joelclimbsthings committed:
Declaring wordpress data package

joelclimbsthings committed:
Allow generation of short description if previous was generated by AI

Commit:
Skip failing tests

Commit:
Changelog

tommyshellberg commented on plugins/woo-ai/src/product-description/product-description-button-container.tsx:
```diff
@@ -189,8 +211,14 @@ export function WriteItForMeButtonContainer() {
 			await requestCompletion( prompt );
+			if ( ! shortTinyEditor.getContent() || shortDescriptionGenerated ) {
+				await requestShortCompletion(
+					`Please provide a brief, 1-2 sentence summary of the following product description in fewer than 50 words:\n ${ tinyEditor.getContent() }`
```
I think we will want to test out some alternative prompts here, specifically, ones that are geared towards writing meta snippets that appear under SERPs. For instance, I got a description with this PR that was too long to fit underneath a search engine result (it should be max 155 characters):

![Screenshot 2023-07-21 at 17 10 06](https://github.com/woocommerce/woocommerce/assets/6723003/1d37aa06-fe4a-478c-95dc-d70a86d6b940)
![Screenshot 2023-07-21 at 17 13 57](https://github.com/woocommerce/woocommerce/assets/6723003/ccbd77aa-26f0-4d99-b664-c1fbd63e6aa2)

It gets cut off when it's too long which leads to awkward results.

I tried writing a longer prompt which seemed to provide decent results:
```
Please write a high-converting Meta Description for the following WooCommerce product description. It should entice someone from a search results page to click on the product link. 
It should be no more than 155 characters so that the entire meta description fits within the space provided by the search engine result without being cut off or truncated. It should explain what users will see if they click on the product page link. It should include the target keyword for the product: {description here}
```

![Screenshot 2023-07-21 at 17 20 14](https://github.com/woocommerce/woocommerce/assets/6723003/41dc5be7-17f1-425b-982f-442a944736f9)

joelclimbsthings replied:
@tommyshellberg Nice, I like it! I actually didn't give the prompt too much thought 😆. I updated this with your prompt with just a couple minor tweaks based on what I observed.

joelclimbsthings committed:
Improving prompt based on feedback

joelclimbsthings committed:
Adding line to prompt to prevent special characters

Commit:
Skip tax rates test if CI

samueljseay committed:
WooCommerce Docs: Add support for more core block conversion (#39244)

* WIP extend support for frontmatter in manifests.

* Change some of the prop names in the frontmatter support, add tests.

* WIP support new frontmatter in plugin.

* Improve category processing.

* restructure the docs to have a nested category with no index.

* Add WP test stubs to dev.

* Add tests for the ManifestProcessor.

* Add param docs to function

* Store post meta on posts when they are updated or created. Add supporting tests.

* WIP supporting more core block types.

* Fix bugs in conversion.

* Adjust conversion for issues with tables, blockquotes, code.

* Resolve pnpm lock conflicts

* Return manifest to state in trunk.

* Return pnpm lock to trunk state.

* Update fixture for new md content.

rjchow committed:
dev: core profiler pages storybook (#39046)

* dev: core profiler pages storybook

* changed css path to pre-build zip path

Commit:
Update/39325 typo in core profiler data sharing copy (#39327)

* Fix grammar in data-sharing agreement copy

* Add changelog

* Put ,

* Add brackets

* Move learn more link inside the bracket

* Update test snapshot

* Fix grammar

* Update test snapshot

github-actions[bot] committed:
Delete changelog files based on PR 39327 (#39388)

Delete changelog files for 39327

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

psealock committed:
Monrepo Utils code-freeze: Update version bump to modify release branches (#39243)

crunnells committed:
Replace 'end' CSS props with 'flex-end' which has broader browser support

psealock committed:
WooCommerce Docs: Handle multiple category posts (#39352)

vedanshujain committed:
Support inserting NULL values for strict DB mode (#39396)

* Support inserting NULL values for strict DB mode

* Set default date for placeholder order to support strict MySQL.

* Add unit test to verify strict mode also works.

* Make HPOS behavior of modified date consistent with WP_Post.

In HPOS we were leaving modified date to be empty, while WP_Post set it to the created date if modified date is null.

jorgeatorres committed:
Include postcode in Vietnam address format

jorgeatorres committed:
Add changelog

coreymckrill committed:
Use the LegacyProxy to invoke get_plugins, remove CodeHacking usage

crunnells committed:
Added changelog

samueljseay committed:
WooCommerce Docs: Fix a bug where passing full file urls made ids unstable. (#39357)

Fix a bug where passing full file urls made ids unstable.

vedanshujain committed:
Use direct post calls for meta CRUD for performance.

vedanshujain committed:
Add changelog.

vedanshujain committed:
Prevent read on sync while inside a backfill.

Commit:
Shard unit test runs (#39362)

PanosSynetos committed:
👌 Use admin theme color for select2

tomalec committed:
Remove non-existing method from TaskList docs.

tomalec committed:
Add the changelog entry.

nathanss committed:
[Product Block Editor] Add post_password parameter to the Woo product REST api (#39438)

* Add post password to API

* Add changelog

* Fix phpcs issue

* Remove post_password from tests

* Add additional property to test

* Increment number of properties in product schema

* Update the post when post_password changes

Commit:
Optimize system status tests (#39363)

coreymckrill committed:
Update template version

psealock committed:
Monorepo Utils: Fix no merge base in changefile script (#39467)

* when checking out remote branch, ensure git history is there to find common ancestor for git diff

* remove unshallow

vedanshujain committed:
Fixup - use correct syntax for removing element.

vedanshujain committed:
Add changelog.

vedanshujain committed:
Use seperate variables for backfilling and sync on read.

Also add explaining comments.

vedanshujain committed:
Account for meta value being array when compared.

vedanshujain committed:
Preserve original meta value.

vedanshujain committed:
Add unit test for adding meta within adding meta.

vedanshujain committed:
Unit test fixup.

vedanshujain committed:
Handle array to non array comparison gracefully.

nathanss committed:
[Product Block Editor] Add require password block field (#39464)

* Add post password to API

* Add changelog

* Fix phpcs issue

* Remove post_password from tests

* Add additional property to test

* Increment number of properties in product schema

* Update the post when post_password changes

* Start adding password block

* Add css and import it

* Refactor attributes and erase password when checkbox is unchecked

* Add changelogs

* Remove unused imports

* Rename 'fields' to 'field'

* Refactor CSS

* Remove example object

nathanss committed:
[ Product Block Editor ] Create Variation options block (#39256)

* Add support for variable products

* Add 'hello world' block to variations tab

* Add product-section block to template

* Add AttributeControl component to screen

* Add changelog

* Change labels

* Make a copy of AttributeControl to VariationOptionsControl to allow the fields to evolve separately in future

* Fix tests

* Add changelog to woocommerce

* Fix alert error

* Remove copied control and start adapting attribute control to handle both scenarios

* Add -field to block name

* Revert "Add -field to block name"

This reverts commit 50e1ee66e27ffb2df22ea7f6a4f78d1577a273f5.

* Revert "Revert "Add -field to block name""

This reverts commit eee0441c6532f9fa8cf8383d9699fd503dd054ae.

* Extract more labels

* Hide drag handle in variation options

AndrewJDawes committed:
Fix phpcs sniff error

AndrewJDawes committed:
Explicitly fallback to returning null

vedanshujain committed:
Add to reading order ID when creating new order.

vedanshujain committed:
Use more granular backfill op.

vedanshujain committed:
Fix indentation/

vedanshujain committed:
Refresh order to account for DB changes from post hooks.

yashitamittal11 committed:
Update task list items to show spinner on selection (#39270)

* Update task list to show a spinner on item click

There is an issue of a noticeable delay when clicking on product task list items, which leaves users waiting without any indication of the loading status.
To enhance the user experience, add a loading spinner that provides visual feedback during the transition to the next page.

Co-authored-by: yashita.mittal <yashita.mittal@a8c.com>

vedanshujain committed:
Fixups from rebase conflicts.

vedanshujain committed:
Fix unit test feature + refresh DB order before backfill.

rrennick committed:
Add changelog file

louwie17 committed:
Pass down height of inserter panel to allow for scrolling (#39491)

* Pass down height of inserter panel to allow for scrolling

* Add changelog

Commit:
Align checkbox, logo, and title on the plugins page (core profiler) (#39394)

* Reset checkbox position

* Center align chekbox, logo, and the title

* Add changelog

* Remove use of classnames -- no longer needed

* Fix CSS lint issues

Commit:
Redirect to Jetpack connect page on plugins page skip (#39109)

* Redirect users to Jetpack Auth when the users deselect all plugins, but Jetpack is already installed

* Add changelog

gigitux committed:
Bump WC Blocks to 10.6.2 (#39492)

bump WC Blocks to 10.6.2

github-actions[bot] committed:
Delete changelog files based on PR 39492 (#39496)

Delete changelog files for 39492

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

Commit:
E2E: Fix and simplify user id retrieval in `a user should see 3 sections by default` (#39472)

* Add changelog

* Simplify user id retrieval

mattsherman committed:
Only set attribute as used for variation by default if product type is variable

Commit:
E2E: Fix and simplify user id retrieval in `a user should see 3 sections by default` (#39472)

* Add changelog

* Simplify user id retrieval

mattsherman committed:
Trigger woocommerce_tab_shown when a tab is shown

Commit:
E2E: Fix and simplify user id retrieval in `a user should see 3 sections by default` (#39472)

* Add changelog

* Simplify user id retrieval

mattsherman committed:
Add the empty attribute when the attributes tab is shown, to account for product type switching

Commit:
E2E: Fix and simplify user id retrieval in `a user should see 3 sections by default` (#39472)

* Add changelog

* Simplify user id retrieval

mattsherman committed:
Do not disable Used for variations checkbox by default

mattsherman committed:
Remove disable_or_enable_fields() implementation

mattsherman committed:
Restore woocommerce_added_attribute trigger

mattsherman committed:
Changelog

mattsherman committed:
Check if product_type query arg is set before using

psealock committed:
WooCommerce Docs: Update manifest hash on content update (#39445)

samueljseay committed:
Remove all schema checking code from Code Analyzer. (#39487)

vedanshujain committed:
Move clear cache statement to its original position.

xristos3490 committed:
Update the admin's menu remaining tasks bubble CSS class and loosen the query selector for handling the runtime update (#39273)

* Modify dom structure, add a specific class and alter the querySelector

* Select all bubbles instead of only the first one

* Add changelog

* Cleanup

* Cleanup

rjchow committed:
add: feature flag for customize your store task (#39397)

* add: feature flag for customize your store task

* toggled core-profiler feature flag to true in dev mode

louwie17 committed:
Dequeue conflicting styles within product editor (#39478)

* Dequeue conflicting styles within product editor

* Add changelog

* Fix lint errors

* Add check to only dequeue on admin/embed pages

github-actions[bot] committed:
Delete changelog files based on PR 39502 (#39511)

Delete changelog files for 39502

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Delete changelog files based on PR 39478 (#39514)

Delete changelog files for 39478

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

coreymckrill committed:
Set the HPOS-related features back to "experimental"

invalid-email-address committed:
Add changefile(s) from automation for the following project(s): woocommerce

jorgeatorres committed:
Prevent possible fatal error when user holds edit lock for a deleted order (#39497)

* Prevent possible fatal error when locked order is deleted

* Add changelog

jorgeatorres committed:
Prevent possible fatal error when refreshing edit lock for orders (#39498)

* Remove return type from signature for order edit lock callbacks

* Add changelog

coreymckrill committed:
Guest verification: Be more precise when checking form submission data (#39479)

Commit:
Use the same checkbox styles on the platform selector (#39469)

* Use the same checkbox style

* Add changelog

* Update plugins/woocommerce-admin/client/core-profiler/components/multiple-selector/multiple-selector.scss

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

* Fix CSS lint issues

---------

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

samueljseay committed:
WooCommerce Docs: Relative Link Parsing (#39293)

louwie17 committed:
Revert "Update product experiment name (#39210)" (#39523)

* Revert "Update product experiment name (#39210)"

This reverts commit c547326135f7cae9ea09f45b42db9f17d8bcfa0a.

* Add changefile(s) from automation for the following project(s): woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

vedanshujain committed:
Update plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

Co-authored-by: Ron Rennick <ron@ronandandrea.com>

jonathansadowski committed:
Prepare trunk for WordPress 6.3 (#39518)

* Update woocommerce.php for WP 6.3

* Update readme.txt for WP 6.3

* Add changefile(s) from automation for the following project(s): woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

louwie17 committed:
Add new variation modal (#39522)

* Hook up add new variations options modal

* Fix duplicate logic and test

* Add changelog

* Match local attributes by name case incentive

* Remove console log

* Make use of some function instead of findIndex

vedanshujain committed:
Add sync and usage flags to legacy features. (#39537)

* Add sync and usage flags to legacy features.

* Use transient instead of constant.

* Add unit test and changelog.

* Add more robustness to unit test by adding more setting values.

* PHPCS fixes

---------

Co-authored-by: Jorge Torres <jorge.torres@automattic.com>

github-actions[bot] committed:
Delete changelog files based on PR 39520 (#39554)

Delete changelog files for 39520

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Delete changelog files based on PR 39537 (#39556)

Delete changelog files for 39537

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

github-actions[bot] committed:
Delete changelog files based on PR 39523 (#39558)

Delete changelog files for 39523

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

gigitux committed:
bump WC Blocks to 10.6.3 (#39560)

jonathansadowski committed:
Update workflow to not add label when PR has already been analyzed (#39545)

mdperez86 committed:
Feedback bar is mis-aligned; transient notices overlap (#39233)

* Fix the feedback bar alignment to be centered

* Move the transient notices to the top of the feedback bar

* Add changelog file

* Remove feedback bar animation

github-actions[bot] committed:
Delete changelog files based on PR 39560 (#39568)

Delete changelog files for 39560

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

nerrad committed:
Adding theme docs structure (#39564)

* Initial theme doc folder structure

* condense to one doc for marketplace-guidelines

Content will be implemented in separate PR.

faisal-alvi committed:
Fix #38716 Detect the first `<h2>` tag to fix Test failure (#38717)

* Fix #38716

* Adding changelog

* update changelog

Commit:
Update/79113 incorrect subheading style and copy on wc core profiler (#39526)

* Use -webkit-font-smoothing: antialiased for core profiler subheadings

* Remove subheading letter-spacing

nigeljamesstevenson committed:
Update locator to reduce flakiness when running tests against an external site (#39375)

* Update locator to reduce flakiness when running tests against an external site

* changelog update

psealock committed:
WooCommerce Docs: Delete posts when it removed from the manifest (#39395)

samueljseay committed:
Beta Tester: Do not run code in tracks debug unless WC_ABSPATH is defined. (#39486)

* Do not run code in tracks debug unless WC_ABSPTAH is defined.

* Add changefile(s) from automation for the following project(s): woocommerce-beta-tester

* Also avoid registering scripts if Woo is not installed.

---------

Co-authored-by: github-actions <github-actions@github.com>

tommyshellberg committed:
Remove CSS that was needed for temporary snackbar fix.

tommyshellberg reviewed (approved):
LGTM! Created a separate issue for remaining feedback:
https://github.com/woocommerce/woocommerce/issues/39580
Also removed a piece of unnecessary SCSS code. Merging.

barryhughes committed:
Remove no-op migration from 8.0.

jorgeatorres committed:
Do not run `WC_Admin_Meta_Boxes::save_meta_boxes()` for HPOS orders

jorgeatorres committed:
Add changelog

barryhughes committed:
Temporarily skip problematic test (tested feature remains experimental).

github-actions[bot] committed:
Delete changelog files based on PR 39584 (#39589)

Delete changelog files for 39584

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

gigitux committed:
bump WooCommerce blocks version to 10.6.4 (#39582)

github-actions[bot] committed:
Delete changelog files based on PR 39582 (#39592)

Delete changelog files for 39582

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

octaedro committed:
Fix selection of term after clicking the value name (#39574)

* Fix select attribute values

* Add changelog

* Move changelog

* Remove 'list-item' class name

joshuatf committed:
Fix toolbar button text not showing in WP 6.3 (#39572)

* Fix toolbar button text not showing in WP 6.3

* Add changelog entry

tommyshellberg committed:
Merge branch 'trunk' into add/update-short-description-39151
